model/map/src/main/java/org/keycloak/models/map/userSession/MapUserSessionProvider.java

```diff
@@ -291,8 +290,17 @@ public UserSessionModel getUserSession(RealmModel realm, String id) {
     @Override
     public Stream<UserSessionModel> getUserSessionsStream(RealmModel realm, ClientModel client,
                                                           Integer firstResult, Integer maxResults) {
-        return paginatedStream(getUserSessionsStream(realm, client)
-                .sorted(Comparator.comparing(UserSessionModel::getLastSessionRefresh)), firstResult, maxResults);
+        LOG.tracef("getUserSessionsStream(%s, %s, %s, %s)%s", realm, client, firstResult, maxResults, getShortStackTrace());
+
+        ModelCriteriaBuilder<UserSessionModel> mcb = realmAndOfflineCriteriaBuilder(realm, false)
+                .compare(UserSessionModel.SearchableFields.CLIENT_ID, ModelCriteriaBuilder.Operator.EQ, client.getId());
+
+
+        return userSessionTx.read(mcb, userSessionStore.createQueryParametersBuilder()
+                    .pagination(firstResult, maxResults, UserSessionModel.SearchableFields.LAST_SESSION_REFRESH)
+                    .build())
+                .map(userEntityToAdapterFunc(realm))
+                .filter(Objects::nonNull);
```

Comment:
Should these two lines be swapped?

Reply:
Probably no. This is there because the `userEntityToAdapterFunc` lambda can return `null` when user session is expired. This is basically copy&paste from other methods that are not using pagination.